docs/en/react/react-lynx-testing-library/index.mdx

```diff
@@ -0,0 +1,624 @@
+  };
+  await waitForElementToBeRemoved(loading);
+  // Since Lynx Environment uses jsdom to implement Element PAPI at the bottom layer
+  // you can directly access document.body to get page elements
+  expect(document.body).toMatchInlineSnapshot(`
+    <body>
+      <page>
+        <text
+          data-testid="message"
+        >
+          Loaded this message:
+          <wrapper>
+            Hello World
+          </wrapper>
+          !
+        </text>
+      </page>
+    </body>
+  `);
+  expect(screen.getByTestId('message')).toHaveTextContent(/Hello World/);
+  expect(elementTree.root).toMatchInlineSnapshot(`
+    <page>
+      <text
+        data-testid="message"
+      >
+        Loaded this message:
+        <wrapper>
+          Hello World
+        </wrapper>
+        !
+      </text>
+    </page>
+  `);
+});
+```
+
+### Rerender
+
+```jsx
+import '@testing-library/jest-dom';
+import { render } from '@lynx-js/react/testing-library';
+import { expect } from 'vitest';
+
+it('rerender will re-render your component', async () => {
+  const Greeting = (props) => <text>{props.message}</text>;
+  const { container, rerender } = render(<Greeting message="hi" />);
+  expect(container).toMatchInlineSnapshot(`
+    <page>
+      <text>
+        hi
+      </text>
+    </page>
+  `);
+  expect(container.firstChild).toHaveTextContent('hi');
+
+  {
+    // Unlike React Testing Library, container needs to be retrieved after rerender
+    // Because ReactLynx creates a new page element each time it loads
+    const { container } = rerender(<Greeting message="hey" />);
+    expect(container.firstChild).toHaveTextContent('hey');
+
+    expect(container).toMatchInlineSnapshot(`
+      <page>
+        <text>
+          hey
+        </text>
+      </page>
+    `);
+  }
+});
+```
+
+### list
+
+> For list usage documentation, please refer to: https://lynxjs.org/zh/api/elements/built-in/list.html
+
+```jsx
+import { useState } from '@lynx-js/react';
+import { render } from '@lynx-js/react/testing-library';
+import { expect } from 'vitest';
+
+it('list', () => {
+  const Comp = () => {
+    const [list, setList] = useState([0, 1, 2]);
+    return (
+      <list>
+        {list.map((item) => (
+          <list-item key={item} item-key={item}>
+            <text>{item}</text>
+          </list-item>
+        ))}
+      </list>
+    );
+  };
+  const { container } = render(<Comp />);
+  expect(container).toMatchInlineSnapshot(`
+    <page>
+      <list
+        update-list-info="[{"insertAction":[{"position":0,"type":"__Card__:__snapshot_f75b7_test_2","item-key":0},{"position":1,"type":"__Card__:__snapshot_f75b7_test_2","item-key":1},{"position":2,"type":"__Card__:__snapshot_f75b7_test_2","item-key":2}],"removeAction":[],"updateAction":[]}]"
+      />
+    </page>
+  `);
+  const list = container.firstChild;
+  // Trigger the componentAtIndex method of list and load the 0th item
+  const uid0 = elementTree.triggerComponentAtIndex(list, 0);
```

Comment:
Before we bring these internal api into public, can we re-consider the naming of it?

At least replace `Component` with `listItem` or something similar :D

Reply:
- `triggerComponentAtIndex` calls `componentAtIndex` of list element
- `triggerEnqueueComponent` calls `enqueueComponent` of list element

However, `componentAtIndex` and `enqueueComponent` are not directly exposed to users.

Let me use `enterListItemAtIndex` and `leaveListItem` for now.

https://github.com/upupming/lynx-stack/blob/97557eb25c3466a6fba8a28713155fa621e9de4f/packages/testing-library/lynx-environment/src/lynx/ElementPAPI.ts#L437